Snakemake/workflow/rules/merge_samples_vcf.smk

```diff
@@ -1,71 +1,99 @@
-# ---------------------------------------------------------------------------- #
-#                               MERGE                                          #
-# ---------------------------------------------------------------------------- #
+import yaml
+vcfdir = "vcfDir"
+chromosomes = [f'Chr{n}' for n in range(1, 3)]
 
-# Starts by extracting a list of samples ids from each vcf.
-# Executes for all chromosomes and all files at once. Outputs temporary .txt
-# for each list.
+mergeList = f'{vcfdir}/Chr1/MergeList.yaml'
+with open(mergeList) as f:
+    files2merge = yaml.safe_load(f)
 
-rule get_samples:
-    input:
-        files=f'{vcfdir}/{{chromosome}}/{{file}}.vcf.gz'
-    output: temp(f'{vcfdir}/{{chromosome}}/{{file}}.txt')
-    conda:
-        'envs/vcfEdit.yaml'
-    log: 'logs/{chromosome}_{file}.log'
-    shell:
-        """
-        bcftools query -l {input.files} > {output}
-        """
+files=[f for f in files2merge.values()][0]
+nfiles2merge = len(files)
+names = [f.strip('.vcf.gz').split('/')[-1].split('_')[-1] for f in files]
 
-# Executes per chromosome! Takes all .txt files at once, compare and remove
-# duplicates, and write a new temporary .txt for each one.
-rule compare:
-    input: [f'{vcfdir}/{{chromosome}}/{{file}}.txt'
-    output: temp([f'{vcfdir}/{{chromosome}}/{{file}}.filtered')
-    log: expand('logs/{{chromosome}}_{file}.log', file=allfiles)
-    run:
-        from itertools import combinations
-        samplelist=[]
-        for ifile in input:
-            with open(ifile, 'r') as f:
-                samplelist.append(f.read().splitlines())
+rule all:
```

Comment:
WE probably also remove this in the next step, right? When we edit the Snakefile

Reply:
removed all rule alls